packages/core/src/render3/instructions/defer.ts

```diff
@@ -510,15 +511,17 @@ function registerDomTrigger(
       cleanup();
     }, injector);
 
-    afterRenderRef.destroy();
     storeLViewOnDestroy(triggerLView, cleanup);
 
     // Since the trigger and deferred block might be in different
     // views, we have to register the callback in both locations.
     if (initialLView !== triggerLView) {
       storeLViewOnDestroy(initialLView, cleanup);
     }
-  }, {injector});
+  }
+
+  // Begin polling for the trigger.
+  beforeNextAfterRender(pollDomTrigger, {injector});
```

Comment:
Just a couple ideas for naming (as a bikeshed :)):

```suggestion
  preAfterNextRenderHook(pollDomTrigger, {injector});
```

```suggestion
  internalAfterNextRender(pollDomTrigger, {injector});
```

I like the `internalAfterNextRender` a bit more, since we retain the "AfterNextRender" term in the name and indicate that this is an internal/special one.